### ISSUE TITLE
feat(CA): more verbose logs for scale-down blocking PDB 

### ISSUE THREAD

**Issue by mattxwang** [kubernetes/autoscaler#4201]
**Which component are you using?**:

The Cluster Autoscaler (CA).

**Is your feature request designed to solve a problem? If so describe the problem this feature should solve.**:

Currently at @adobe I'm working with the CA, investigating reasons why it might not scale-in a node properly and resolving them when possible. In my manual investigation, I've been relying on log messages generated from [`cluster-autoscaler/utils/drain.go::GetPodsForDeletionOnNodeDrain()`](https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/utils/drain/drain.go#L76) and [`cluster-autoscaler/simulator/drain.go::checkPdbs()`](https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/simulator/drain.go#L103), like:

```
Fast evaluation: node XXX cannot be removed: not enough pod disruption budget to move YYY
```

In the next step, I want to answer questions like:

* what's the current pdb for XXX?
* how much budget is needed to move YYY?
* in general, are there any noteworthy things about pod YYY? for example, could it be in a CrashLoopBackOff state? (this might be scope-creep)

However, I'd like to be able to do it rapidly - and, if there are many nodes being blocked by many pods, manually (or even programmatically) investigating these is non-trivial.

**Describe the solution you'd like.**:

I'd like to log more information about the relevant nodes and pods, which could be opt-in/opt-out via log verbosity. For example, 

```
Fast evaluation: node XXX cannot be removed: not enough pod disruption budget to move YYY. The ReplicaSet AAA has BBB current replicas, but desires CCC.
```

Or, blending together the `kubectl describe rs/*` output of a backing ReplicaSet:

```
...
Replicas:       3 current / 3 desired
Pods Status:    3 Running / 0 Waiting / 0 Succeeded / 0 Failed
...
```

I'm unfamiliar with the logging convention and of course am wiling to defer to those who know more!

**Describe any alternative solutions you've considered.**:

Not sure (other than manual testing), but open to suggestions!

**Additional context.**:

One related issue that I'm working on is when the PDB prevents scale-down, but the app "obviously" won't meet the criteria soon - ex, there are zero active pods because of continual CrashLoopBackOff, it is requesting 25 replicas, and requires a min of 3 replicas; I think we should be able to scale-in some of the resources without affecting the ability to fulfill the actual PDB restriction. Completely out of scope of the issue, but just some extra context!

**Comment by mattxwang** [comment]
Don't work at Adobe anymore, but it's possible that someone there is still interested in this. @jantzenallphin would you know?
